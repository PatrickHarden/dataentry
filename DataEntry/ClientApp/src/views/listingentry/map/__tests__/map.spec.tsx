import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Map from '../map';

describe('components', () => {
  describe('MapView', () => {
    let props: any;
    let wrapper: any;
    let store: any;

    const initialState = {
      system: {
        configDetails: {
          loaded: true,
          config: {
          }
        }
      },
      mapping: {
        country: {
        }
      }
    };
    const mockStore = configureStore();
  
    beforeEach(() => {
      props = {
        coordinates: {
          lat:0,
          lng:0
        },
        validations: {
          'field1': {required: true}
        }
      }
      store = mockStore(initialState);
    });

    it("check the existence of lat tag inside the map component", () => {
      wrapper = mount(<Provider store={store}><Map {...props} /></Provider>);
      expect(wrapper.exists('[name="lat"]')).toBeTruthy()
    });

    it("check the existence of lng tag inside the map component", () => {
      wrapper = mount(<Provider store={store}><Map {...props} /></Provider>);
      expect(wrapper.exists('[name="lng"]')).toBeTruthy()
    });

    it("given a prop the value is set", () => {
      const lat = "10";
      const lng = "20";
      props.coordinates.lat = lat;
      props.coordinates.lng = lng;
      wrapper = mount(<Provider store={store}><Map {...props} /></Provider>);
      expect(wrapper.exists('[id="lat"][defaultValue="' + lat + '"]')).toBeTruthy();
      expect(wrapper.exists('[id="lng"][defaultValue="' + lng + '"]')).toBeTruthy();
    });

    // snapshot render
    it('renders map to snapshot', () => {
      expect(shallow(<Provider store={store}><Map {...props} /></Provider>)).toMatchSnapshot();
    });
  });
});