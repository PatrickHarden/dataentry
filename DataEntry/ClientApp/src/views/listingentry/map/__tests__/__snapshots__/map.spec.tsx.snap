// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components MapView renders map to snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Memo(Map)
    coordinates={
      Object {
        "lat": 0,
        "lng": 0,
      }
    }
    validations={
      Object {
        "field1": Object {
          "required": true,
        },
      }
    }
  />
</ContextProvider>
`;
